File: server/app/schemas/application.py
from pydantic import BaseModel
from typing import Optional, Literal
from uuid import UUID


class ApplyRequest(BaseModel):
    cover_letter: Optional[str] = None


class ApplyResponse(BaseModel):
    message: str
    application_id: UUID


class ApplicationStatusResponse(BaseModel):
    has_applied: bool
    application_id: Optional[UUID] = None
    status: Optional[str] = None


class ApplicationListItem(BaseModel):
    application_id: UUID
    applicant_id: UUID
    applicant_name: str
    resume_url: Optional[str] = None
    status: Literal["applied", "under review", "shortlisted", "rejected", "hired"]


class UpdateApplicationStatusRequest(BaseModel):
    status: Literal["applied", "under review", "shortlisted", "rejected", "hired"]


class UpdateApplicationStatusResponse(BaseModel):
    message: str

File: client/web/src/types/application.ts
export type ApplicationStatus = "applied" | "under review" | "shortlisted" | "rejected" | "hired"

export type JobApplicationItem = {
  application_id: string
  applicant_id: string
  applicant_name: string
  applicant_email: string
  resume_url: string | null
  status: ApplicationStatus
}

export type MyApplicationItem = {
  application_id: string
  status: ApplicationStatus
  job_id: string
  job_title: string
  job_type: "full-time" | "part-time" | "internship" | "contract"
  job_location: string
  company_id: string
  company_name: string
}